rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala

```diff
@@ -39,35 +41,45 @@ trait HotStoreSpec[F[_], M[_]] extends FlatSpec with Matchers with GeneratorDriv
 
   implicit val arbitraryJoins = distinctListOf[Join].map(_.toVector)
 
-  implicit def arbitraryCache: Arbitrary[Cache[String, Pattern, String, StringsCaptor]] = Arbitrary(
-    for {
-      continuations <- Arbitrary.arbitrary[TrieMap[Seq[String], Seq[
-                        WaitingContinuation[Pattern, StringsCaptor]
-                      ]]]
-      installedContinuations <- Arbitrary.arbitrary[
-                                 TrieMap[Seq[String], WaitingContinuation[Pattern, StringsCaptor]]
-                               ]
-      data           <- Arbitrary.arbitrary[TrieMap[String, Seq[Datum[String]]]]
-      joins          <- Arbitrary.arbitrary[TrieMap[String, Seq[Seq[String]]]]
-      installedJoins <- Arbitrary.arbitrary[TrieMap[String, Seq[Seq[String]]]]
-    } yield Cache(
-      continuations,
-      installedContinuations,
-      data,
-      joins,
-      installedJoins
+  implicit def arbitraryHotStoreState
+      : Arbitrary[HotStoreState[String, Pattern, String, StringsCaptor]] =
+    Arbitrary(
+      for {
+
+        continuations <- Arbitrary
+                          .arbitrary[TrieMap[Seq[String], Seq[
+                            WaitingContinuation[Pattern, StringsCaptor]
+                          ]]]
+                          .map(_.toMap)
+        installedContinuations <- Arbitrary
+                                   .arbitrary[
+                                     TrieMap[
+                                       Seq[String],
+                                       WaitingContinuation[Pattern, StringsCaptor]
+                                     ]
+                                   ]
+                                   .map(_.toMap)
+        data           <- Arbitrary.arbitrary[TrieMap[String, Seq[Datum[String]]]].map(_.toMap)
+        joins          <- Arbitrary.arbitrary[TrieMap[String, Seq[Seq[String]]]].map(_.toMap)
+        installedJoins <- Arbitrary.arbitrary[TrieMap[String, Seq[Seq[String]]]].map(_.toMap)
```

Comment:
`immutable.Map` can be used here instead of TrieMap.

Reply:
The whole method is not used actually . I just removed it .